site/zenodo_rdm/subcommunities/request.py

```diff
@@ -113,8 +117,111 @@ class ZenodoSubCommunityRequest(SubCommunityRequest):
+            "All current and future records in your community will be automatically indexed "
+            "in the EU Open Research Repository increasing their visibility</li><li><b>Curation:</b> "
+            "All records will be subject to the EU Open Research Repository "
+            "<a href='https://zenodo.org/communities/eu/curation-policy'>curation policy</a>. "
+            "For instance, you can only deposit records in the community related to the EU-funded "
+            "project</li><li><b>Verified:</b> All EU project communities are marked with a "
+            "Verified community badge</li></ul></b></br>The EU Open Research Repository is gradually "
+            "being improved and by mid-2025 new submissions will automatically be checked "
+            "for compliance with the related open science requirements in the Horizon Europe "
+            "grant agreement. For more information see <a href='https://zenodo.org/communities/eu/pages/join'>"
+            "https://zenodo.org/communities/eu/pages/join</a>.</br></br><b>Which EU-funded projects "
+            "have already joined?</b></br>You can browse the "
+            "<a href='https://zenodo.org/communities/eu/browse/subcommunities'>EU-funded projects</a> "
+            "which have already already joined.</br></br><b>When should I decline the invitation "
+            "to join?</b></br>You should <b>decline</b> this invitation if your Zenodo community "
+            "is not related to the above mentioned EU-funded project, or if the community is used "
+            "for multiple purposes (e.g both an organisation and a project).</br></br><b>"
+            "Further questions?</b></br>Don't hesitate to get in <a href='https://zenodo.org/support'>"
+            "touch with us</a> if you have any questions.</br></br>The request will be automatically "
+            f"accepted on <b>{expires_at}</b> in case you do not accept or decline the request by then."
+            "</br></br>Your sincerely,</br>The Zenodo team"
+        )
+
+        super().execute(identity, uow)
+
+
+class SubCommunityInvitationAcceptAction(AcceptSubcommunityInvitation):
+    """Represents an accept action used to accept a subcommunity.
+
+    Zenodo re-implementation of the accept action, to also move the records.
+    """
+
+    def execute(self, identity, uow):
+        """Execute approve action."""
+        child = self.request.receiver.resolve().id
+        parent = self.request.created_by.resolve().id
+
+        _add_community_records(child, parent, uow)
+        # moving the community is handled by super()
+
+        super().execute(identity, uow)
+
+
+class SubcommunityInvitationExpireAction(actions.ExpireAction):
+    """Expire action."""
+
+    def execute(self, identity, uow):
+        """Execute expire action."""
+        child = self.request.receiver.resolve().id
+        parent = self.request.created_by.resolve().id
+
+        _add_community_records(child, parent, uow)
```

Comment:
does it mean you add records to parent community when invitation expires? Since I don't know the requirements it seems a bit counter intuitive.

Reply:
yep, by default they will be included if the request expires. The thinking is that this is mutually beneficial and the request would only be declined if there has been a mistake